flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/TableEnvironment.scala

```diff
@@ -152,13 +160,13 @@ abstract class TableEnvironment(val config: TableConfig) {
     *
     * @param name The name under which the table is registered.
     * @param table The table to register in the catalog
-    * @throws TableException if another table is registered under the provided name.
+    * @throws ValidationException if another table is registered under the provided name.
     */
-  @throws[TableException]
+  @throws[ValidationException]
   protected def registerTableInternal(name: String, table: AbstractTable): Unit = {
 
     if (isRegistered(name)) {
-      throw new TableException(s"Table \'$name\' already exists. " +
+      throw new ValidationException(s"Table \'$name\' already exists. " +
```

Comment:
It makes sense to me.
In this way, unsupported operations for StreamTableEnvironment should throws `TableException` ?